src/ruby/ext/grpc/extconf.rb

```diff
@@ -69,6 +69,10 @@
 
 unless windows
   puts 'Building internal gRPC into ' + grpc_lib_dir
+
+  system('git submodule update --init')
```

Comment:
Unfortunately I think this comes with a significant cost, at least specifically for our tests. `git submodule update --init` needs to run over the network and can increase slowness and flakiness of our tests, which build the ruby extension using this same `extconf.rb` file, through an invocation of `rake`.

Why exactly are you referring to the ruby gem through that github url?

Reply:
I want to use https://github.com/grpc/grpc/pull/15356 in my environment. But it seems not to be  released yet. and I don't want to wait until it will be relased. So I referring commit hash.

Reply:
The next branch cut is planned for this Friday BTW, and once the v1.13.x branch is cut, we'll do a ruby pre-release shortly after. Can it wait for that?

Otherwise, note you can build packages locally, and from master, by running `rake gem:native` from the github repo root. Can that work for you?

I'd be open to an easier way to build from master, but I think this adds to much of a burden to `rake compile` (also I'm not sure if this will work with the source-only distributions of the grpc-ruby gem)

Reply:
> Otherwise, note you can build packages locally, and from master, by running rake gem:native from the github repo root.

It would work… but I think it's not a good way to use gem.

IMO, I think that it is better for grpc-ruby gem to build from any branches and commits. Because we have to wait for new features for a long time (one or two weeks), It is not good for developers  and users of grpc-ruby gem.

>  I think this adds to much of a burden to rake compile

When submodule are already updated, I think `git submodule update` doesn't fetch any more data. In any case, we need to `git submodule udpate` to build grpc-ruby, I think it's not a cost. Would you tell me what a cost is.

Reply:
From what I've seen, `git submodule update` <i>does</i> reach out to the network even if everything actually is up to date locally, which is what introduces this problem.

Also though, keep in mind that for source distribution packages, all source code, including submodules, comes bundled, and it would actually be <i>incorrect</i> to update the submodules in that case, since the package is meant to be used with them at a pinned version

Reply:
When doing a `gem install grpc --platform ruby`, I don't think we want to do a `git submodule update --init` call while building.

I'd argue that specifying a commit hash when depending on grpc from master is what we want anyways though - master is not necessarily stable, so depending on a certain commit hash can provide what's needed without the risk of picking up a bad commit.

Reply:
@ganmacs I took a look at what bundler currently provides, since I was thinking that this really sounded like a short coming of bundler's `git` dependency system. However, from the [bundler docks](https://bundler.io/v1.12/git.html), it looks like bundler does in fact allow one to expand submodules in their dependency declaration!

I think what this PR is trying to accomplish can instead be fixed with:

```
gem 'grpc', :git => 'git://github.com/grpc/grpc.git', :submodules => true
```

in your gemspec

Reply:
Great! It's really what I wanted. I'd like to appreciate it. 
I'm closing this PR. Thanks :shipit: